pkg/scheduler/framework/interface.go

```diff
@@ -160,6 +162,29 @@ func (s *Status) AsError() error {
 	return errors.New(s.Message())
 }
 
+// StatusEqual returns whether two Statuses are equal to each other.
+func StatusEqual(x, y *Status) bool {
+	if x == y {
+		return true
+	}
+	if x == nil || y == nil {
+		return x.IsSuccess() && y.IsSuccess()
+	}
+	if x.code != y.code {
+		return false
+	}
+	if x.code == Error {
+		return cmp.Equal(x.err, y.err, cmpopts.EquateErrors())
+	}
+	return len(x.reasons) == 0 && len(y.reasons) == 0
+}
+
+// Equal checks equality of two statuses. This is useful for testing with
+// cmp.Equal.
+func (s *Status) Equal(x *Status) bool {
+	return StatusEqual(s, x)
```

Comment:
why not have the implementation here?

Reply:
@gavinfish

Reply:
I left StatusEqual there so that we could still use cmp.Diff() somewhere. But as we discussed at https://github.com/kubernetes/kubernetes/pull/98481#discussion_r568105785, if we all agree to just use cmp.Equal(), then I will move the implementation here.

Reply:
I think you are misunderstanding. `cmp.Diff` would still use `Status.Equal`. The problem is that it won't be able to tell internal differences (it will just say: this entire status is different from this other one). And it's the same if you use a comparer in `Diff`.

Reply:
oooh, you are right, `cmp.Diff` just depends on `cmp.Equal` to tell whether two objects are equal.

Reply:
`cmp.Comparer` has higher priority but `(s *Status) Equal(x *Status) bool` is better way to define the default equal comparison.

Reply:
It makes sense that a comparer has higher priority, for the cases where someone wants to have a custom comparison.
As for our default comparison, `Status.Equal` is enough.